### ISSUE TITLE
Mac OS #include <version> vs $(ocamlc -where)/lib/ocaml/version

### ISSUE THREAD

**Issue by ygrek** [ocaml/ocaml#10410]
With latest update of OS X we were having weird build problems in various opam packages with C stubs, e.g. : 
```
#=== ERROR while compiling re2.v0.14.0 ========================================#
# context     2.0.7 | macos/x86_64 | ocaml-base-compiler.4.11.2 | git+ssh://private-opam-repository
# path        ~/work/frontend/_opam/.opam-switch/build/re2.v0.14.0
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p re2 -j 3
# exit-code   1
# env-file    ~/.opam/log/re2-3221-1a64be.env
# output-file ~/.opam/log/re2-3221-1a64be.out
### output ###
# # It must be in the format described in stdlib/sys.mli
# [...]
# In file included from stubs.cpp:1:
# In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/vector:276:
# In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__bit_reference:15:
# In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/algorithm:643:
# In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/memory:671:
# In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/new:93:
# /Users/user/work/frontend/_opam/lib/ocaml/version:1:1: error: expected unqualified-id
# 4.11.2
# ^
# fatal error: too many errors emitted, stopping now [-ferror-limit=]
# 20 errors generated.
```

It was tracked down to the combination of these factors:
- presence of `$(ocamlc -where)/lib/ocaml/version` file (notice: lowercase, unlike VERSION in git, but contents is the same)
- `version` file being included somewhere in C++ standard liibrary on OS X e.g. (results of grep) : 
```
/usr/local/Cellar/gcc@10/10.3.0/include/c++/10.3.0/x86_64-apple-darwin20/bits/stdc++.h:148:#include <version>
/usr/local/Cellar/llvm@9/9.0.1_2/include/c++/v1/ostream:143:#include <version>
/usr/local/Cellar/llvm@9/9.0.1_2/include/c++/v1/forward_list:186:#include <version>
/usr/local/Cellar/llvm@9/9.0.1_2/include/c++/v1/unordered_map:414:#include <version>
/usr/local/Cellar/llvm@9/9.0.1_2/include/c++/v1/locale:189:#include <version>
```

idk how uppercase `VERSION` becomes lowercase `version` (maybe opam is involved?), but the situation seems quite fragile

**Comment by Octachron** [comment]
This should be fixed in 4.12.0 :  #9895 .

**Comment by ygrek** [comment]
I see, thanks
